File: src/main/java/com/bing/bingaicode/service/UserService.java
package com.bing.bingaicode.service;

import com.bing.bingaicode.model.dto.UserQueryRequest;
import com.bing.bingaicode.model.vo.LoginUserVO;
import com.bing.bingaicode.model.vo.UserVO;
import com.mybatisflex.core.query.QueryWrapper;
import com.mybatisflex.core.service.IService;
import com.bing.bingaicode.model.entity.User;
import jakarta.servlet.http.HttpServletRequest;

import java.util.List;

/**
 * 用户 服务层。
 *
 * @author bing
 */
public interface UserService extends IService<User> {

    /**
     * 用户注册
     *
     * @param userAccount   用户账户
     * @param userPassword  用户密码
     * @param checkPassword 校验密码
     * @return 新用户 id
     */
    long userRegister(String userAccount, String userPassword, String checkPassword);

    /**
     * 获取脱敏用户信息
     *
     * @param user 用户
     * @return 脱敏用户信息
     */
    LoginUserVO getLoginUserVO(User user);

    /**
     * 用户登录
     *
     * @param userAccount  用户账户
     * @param userPassword 用户密码
     * @return 登录后的用户信息
     */
    LoginUserVO userLogin(String userAccount, String userPassword, HttpServletRequest request);
    /**
     * 获取脱敏后的用户信息
     *
     * @param user 用户信息
     * @return
     */
    UserVO getUserVO(User user);
    /**
     * 获取脱敏后用户信息
     *
     * @param userList 用户
     * @return 脱敏后用户信息
     */
    List<UserVO> getUserVOList(List<User> userList);
    /**
     * 用户注销
     *
     * @param request
     * @return 退出登录是否成功
     */
    boolean userLogout(HttpServletRequest request);
    /**
     * 获取当前登录用户
     *
     * @param request 请求
     * @return 当前登录用户
     */
    User getLoginUser(HttpServletRequest request);
    /**
     * 根据查询条件构造数据查询参数
     *
     * @param userQueryRequest
     * @return
     */
    QueryWrapper getQueryWrapper(UserQueryRequest userQueryRequest);

    /**
     * 获取加密密码
     *
     * @param userPassword 用户密码
     * @return 加密后的密码
     */
    String getEncryptPassword(String userPassword);
}

File: bing-ai-code-backend/app/services/user_service.py
import hashlib
from typing import Optional
from sqlalchemy import select, func, and_
from sqlalchemy.ext.asyncio import AsyncSession

from app.models.user import User
from app.schemas.user import LoginUserVO, UserVO, UserQueryRequest
from app.exceptions import BusinessException, ErrorCode
from app.schemas.common import PageResult


def get_encrypt_password(user_password: str) -> str:
    """MD5("bing" + password) 兼容 Java 版密码哈希"""
    salt = "bing"
    return hashlib.md5((salt + user_password).encode()).hexdigest()


def get_login_user_vo(user: User) -> LoginUserVO:
    return LoginUserVO(
        id=user.id,
        userAccount=user.userAccount,
        userName=user.userName,
        userAvatar=user.userAvatar,
        userProfile=user.userProfile,
        userRole=user.userRole,
        createTime=user.createTime,
        updateTime=user.updateTime,
    )


def get_user_vo(user: User) -> UserVO:
    return UserVO(
        id=user.id,
        userAccount=user.userAccount,
        userName=user.userName,
        userAvatar=user.userAvatar,
        userProfile=user.userProfile,
        userRole=user.userRole,
        createTime=user.createTime,
    )


def get_user_vo_list(users: list[User]) -> list[UserVO]:
    return [get_user_vo(u) for u in users]


async def user_register(
    db: AsyncSession,
    user_account: str,
    user_password: str,
    check_password: str,
) -> int:
    if not user_account or not user_password or not check_password:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "参数不能为空")
    if len(user_account) < 4:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "用户账号过短")
    if len(user_password) < 8 or len(check_password) < 8:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "用户密码过短")
    if user_password != check_password:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "两次输入的密码不一致")

    # 检查账号是否存在
    result = await db.execute(
        select(func.count()).select_from(User).where(
            User.userAccount == user_account, User.isDelete == 0
        )
    )
    if result.scalar() > 0:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "用户账号已存在")

    # 加密密码
    encrypt_password = get_encrypt_password(user_password)

    # 创建用户
    user = User(
        userAccount=user_account,
        userPassword=encrypt_password,
        userRole="user",
    )
    db.add(user)
    await db.flush()
    await db.refresh(user)
    return user.id


async def user_login(
    db: AsyncSession,
    user_account: str,
    user_password: str,
) -> tuple[User, LoginUserVO]:
    if not user_account or not user_password:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "参数不能为空")
    if len(user_account) < 4:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "用户账号过短")
    if len(user_password) < 8:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "用户密码过短")

    encrypt_password = get_encrypt_password(user_password)
    result = await db.execute(
        select(User).where(
            User.userAccount == user_account,
            User.userPassword == encrypt_password,
            User.isDelete == 0,
        )
    )
    user = result.scalar_one_or_none()
    if not user:
        raise BusinessException(ErrorCode.PARAMS_ERROR, "用户不存在或密码错误")

    return user, get_login_user_vo(user)


async def get_login_user_from_session(
    db: AsyncSession,
    session_user_data: dict,
) -> User:
    """从 session 数据获取完整用户"""
    if not session_user_data:
        raise BusinessException(ErrorCode.NOT_LOGIN_ERROR)

    user_id = session_user_data.get("id")
    if not user_id:
        raise BusinessException(ErrorCode.NOT_LOGIN_ERROR)

    result = await db.execute(
        select(User).where(User.id == user_id, User.isDelete == 0)
    )
    user = result.scalar_one_or_none()
    if not user:
        raise BusinessException(ErrorCode.NOT_LOGIN_ERROR)
    return user


def build_user_query_filters(query_request: UserQueryRequest):
    """构建用户查询过滤条件"""
    filters = []
    if query_request.id is not None:
        filters.append(User.id == query_request.id)
    if query_request.userRole is not None:
        filters.append(User.userRole == query_request.userRole)
    if query_request.userAccount is not None:
        filters.append(User.userAccount.like(f"%{query_request.userAccount}%"))
    if query_request.userName is not None:
        filters.append(User.userName.like(f"%{query_request.userName}%"))
    if query_request.userProfile is not None:
        filters.append(User.userProfile.like(f"%{query_request.userProfile}%"))
    return filters
